### ISSUE TITLE
When kong 1.0.0 migrations bootstrap with two-node cassandra database cluster, its got some errors .

### ISSUE THREAD

**Issue by JustUse** [Kong/kong#4226]
Hi!  In fact, many errors were reported at the kong of migrations database, but I ignored them because the initialization was successful at last.(as follows)

[kong@localhost apache-cassandra-3.11.3]$ kong start
database needs bootstrapping; run 'kong migrations bootstrap'
Error: /usr/local/share/lua/5.1/kong/cmd/start.lua:46: nginx: [error] init_by_lua error: /usr/local/share/lua/5.1/kong/init.lua:281: database needs bootstrap; run 'kong migrations bootstrap'
stack traceback:
        [C]: in function 'error'
        /usr/local/share/lua/5.1/kong/init.lua:281: in function 'init'
        init_by_lua:3: in main chunk

  Run with --v (verbose) or --vv (debug) for more details

[kong@localhost apache-cassandra-3.11.3]$ kong  migrations bootstrap
bootstrapping database...
migrating core on keyspace 'kong'...
core migrated up to: 000_base (executed)
Error: [Cassandra error] cluster_mutex callback threw an error: /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:68: [Cassandra error] failed to run migration '001_14_to_15' teardown: ...l/share/lua/5.1/kong/db/migrations/core/001_14_to_15.lua:299: [Read timeout] Operation timed out - received only 0 responses.
stack traceback:
        [C]: in function 'assert'
        ...l/share/lua/5.1/kong/db/migrations/core/001_14_to_15.lua:299: in function <...l/share/lua/5.1/kong/db/migrations/core/001_14_to_15.lua:264>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/db/init.lua:553: in function 'run_migrations'
        /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:68: in function </usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:67>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/db/init.lua:390: in function 'cluster_mutex'
        /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:67: in function 'bootstrap'
        /usr/local/share/lua/5.1/kong/cmd/migrations.lua:114: in function 'cmd_exec'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:87>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:44>
        /usr/local/bin/kong:7: in function 'file_gen'
        init_worker_by_lua:47: in function <init_worker_by_lua:45>
        [C]: in function 'xpcall'
        init_worker_by_lua:54: in function <init_worker_by_lua:52>
stack traceback:
        [C]: in function 'assert'
        /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:68: in function </usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:67>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/db/init.lua:390: in function 'cluster_mutex'
        /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:67: in function 'bootstrap'
        /usr/local/share/lua/5.1/kong/cmd/migrations.lua:114: in function 'cmd_exec'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:87>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:44>
        /usr/local/bin/kong:7: in function 'file_gen'
        init_worker_by_lua:47: in function <init_worker_by_lua:45>
        [C]: in function 'xpcall'
        init_worker_by_lua:54: in function <init_worker_by_lua:52>

  Run with --v (verbose) or --vv (debug) for more details

[kong@localhost apache-cassandra-3.11.3]$ kong  migrations up
Error: [Cassandra error] cluster_mutex callback threw an error: /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:139: database has pending migrations; run 'kong migrations finish'
stack traceback:
        [C]: in function 'error'
        /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:139: in function </usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:135>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/db/init.lua:390: in function 'cluster_mutex'
        /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:135: in function 'up'
        /usr/local/share/lua/5.1/kong/cmd/migrations.lua:135: in function 'cmd_exec'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:87>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:44>
        /usr/local/bin/kong:7: in function 'file_gen'
        init_worker_by_lua:47: in function <init_worker_by_lua:45>
        [C]: in function 'xpcall'
        init_worker_by_lua:54: in function <init_worker_by_lua:52>

  Run with --v (verbose) or --vv (debug) for more details

[kong@localhost apache-cassandra-3.11.3]$ kong migrations finish
migrating core on keyspace 'kong'...
core migrated up to: 001_14_to_15 (executed)
1 migration processed
1 executed

[kong@localhost apache-cassandra-3.11.3]$ kong  migrations up
migrating core on keyspace 'kong'...
core migrated up to: 002_15_to_1 (pending)
migrating oauth2 on keyspace 'kong'...
oauth2 migrated up to: 000_base_oauth2 (executed)
oauth2 migrated up to: 001_14_to_15 (pending)
oauth2 migrated up to: 002_15_to_10 (pending)
migrating acl on keyspace 'kong'...
acl migrated up to: 000_base_acl (executed)
acl migrated up to: 001_14_to_15 (pending)
migrating jwt on keyspace 'kong'...
jwt migrated up to: 000_base_jwt (executed)
jwt migrated up to: 001_14_to_15 (executed)
migrating basic-auth on keyspace 'kong'...
basic-auth migrated up to: 000_base_basic_auth (executed)
basic-auth migrated up to: 001_14_to_15 (executed)
migrating key-auth on keyspace 'kong'...
key-auth migrated up to: 000_base_key_auth (executed)
key-auth migrated up to: 001_14_to_15 (executed)
migrating rate-limiting on keyspace 'kong'...
rate-limiting migrated up to: 000_base_rate_limiting (executed)
rate-limiting migrated up to: 001_14_to_15 (executed)
rate-limiting migrated up to: 002_15_to_10 (pending)
migrating hmac-auth on keyspace 'kong'...
hmac-auth migrated up to: 000_base_hmac_auth (executed)
hmac-auth migrated up to: 001_14_to_15 (executed)
migrating response-ratelimiting on keyspace 'kong'...
response-ratelimiting migrated up to: 000_base_response_rate_limiting (executed)
response-ratelimiting migrated up to: 001_14_to_15 (executed)
response-ratelimiting migrated up to: 002_15_to_10 (pending)
20 migrations processed
14 executed
6 pending

[kong@localhost apache-cassandra-3.11.3]$ kong start
Kong started
[kong@localhost apache-cassandra-3.11.3]$

_Originally posted by @JustUse in https://github.com/Kong/kong/issues/4193#issuecomment-454312409_

**Comment by JustUse** [comment]
> Try "kong migrations bootstrap --db-timeout 60", I am curious if rather than letting it default you hardset it as I saw success vs failures when I left it off. Same for "kong migrations finish --db-timeout 60"
> 
> Do note bootstrap should only be for new keyspace creation. finish is what you run only after upgrading from 0.14.1 to 1.0.2.

I also feel that the database cluster needs to be optimized for configuration, such as timeout. I try "kong migrations bootstrap --db-timeout 60", but also got same errors....


[kong@localhost apache-cassandra-3.11.3]$ kong migrations bootstrap --db-timeout                                                                                                                                                              60
bootstrapping database...
migrating core on keyspace 'kong'...
core migrated up to: 000_base (executed)
core migrated up to: 001_14_to_15 (executed)
core migrated up to: 002_15_to_1 (executed)
migrating oauth2 on keyspace 'kong'...
oauth2 migrated up to: 000_base_oauth2 (executed)
Error: [Cassandra error] cluster_mutex callback threw an error: /usr/local/share                                                                                                                                                             /lua/5.1/kong/cmd/utils/migrations.lua:68: [Cassandra error] failed to run migra                                                                                                                                                             tion '001_14_to_15' teardown: [Read timeout] Operation timed out - received only                                                                                                                                                              0 responses.
stack traceback:
        [C]: in function 'assert'
        /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:68: in function <                                                                                                                                                             /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:67>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/db/init.lua:390: in function 'cluster_mute                                                                                                                                                             x'
        /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:67: in function '                                                                                                                                                             bootstrap'
        /usr/local/share/lua/5.1/kong/cmd/migrations.lua:118: in function 'cmd_e                                                                                                                                                             xec'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/s                                                                                                                                                             hare/lua/5.1/kong/cmd/init.lua:87>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/s                                                                                                                                                             hare/lua/5.1/kong/cmd/init.lua:44>
        /usr/local/bin/kong:7: in function 'file_gen'
        init_worker_by_lua:49: in function <init_worker_by_lua:47>
        [C]: in function 'xpcall'
        init_worker_by_lua:56: in function <init_worker_by_lua:54>

  Run with --v (verbose) or --vv (debug) for more details

**Comment by jeremyjpj0916** [comment]
Strange, I am just a community member but I have never been able to reproduce that. You will probably want to run in --vv mode and also give Kong team(I am just a community member) all your C* configurations and connection settings. Something seems off here.

**Comment by JustUse** [comment]
> 奇怪，我只是一个社区成员，但我从来没有能够重现这一点。您可能希望以-vv模式运行，并且还为Kong团队（我只是社区成员）提供所有C *配置和连接设置。这里似乎有些东西。

[root@localhost kong]# kong migrations bootstrap --db-timeout 60 --vv
2019/01/23 11:21:42 [verbose] Kong: 1.0.0
2019/01/23 11:21:42 [debug] ngx_lua: 10013
2019/01/23 11:21:42 [debug] nginx: 1013006
2019/01/23 11:21:42 [debug] Lua: LuaJIT 2.1.0-beta3
2019/01/23 11:21:42 [verbose] reading config file at /etc/kong/kong.conf
2019/01/23 11:21:42 [debug] reading environment variables
2019/01/23 11:21:42 [debug] admin_access_log = "logs/admin_access.log"
2019/01/23 11:21:42 [debug] admin_error_log = "logs/error.log"
2019/01/23 11:21:42 [debug] admin_listen = {"0.0.0.0:8001","0.0.0.0:8444 ssl"}
2019/01/23 11:21:42 [debug] anonymous_reports = false
2019/01/23 11:21:42 [debug] cassandra_consistency = "ONE"
2019/01/23 11:21:42 [debug] cassandra_contact_points = {"192.168.0.119","192.168.0.146"}
2019/01/23 11:21:42 [debug] cassandra_data_centers = {"dc1:2","dc2:3"}
2019/01/23 11:21:42 [debug] cassandra_keyspace = "kong"
2019/01/23 11:21:42 [debug] cassandra_lb_policy = "RequestRoundRobin"
2019/01/23 11:21:42 [debug] cassandra_port = 9042
2019/01/23 11:21:42 [debug] cassandra_repl_factor = 1
2019/01/23 11:21:42 [debug] cassandra_repl_strategy = "SimpleStrategy"
2019/01/23 11:21:42 [debug] cassandra_schema_consensus_timeout = 20000
2019/01/23 11:21:42 [debug] cassandra_ssl = false
2019/01/23 11:21:42 [debug] cassandra_ssl_verify = false
2019/01/23 11:21:42 [debug] cassandra_timeout = 10000
2019/01/23 11:21:42 [debug] cassandra_username = "kong"
2019/01/23 11:21:42 [debug] client_body_buffer_size = "8k"
2019/01/23 11:21:42 [debug] client_max_body_size = "0"
2019/01/23 11:21:42 [debug] client_ssl = false
2019/01/23 11:21:42 [debug] database = "cassandra"
2019/01/23 11:21:42 [debug] db_cache_ttl = 0
2019/01/23 11:21:42 [debug] db_resurrect_ttl = 30
2019/01/23 11:21:42 [debug] db_update_frequency = 5
2019/01/23 11:21:42 [debug] db_update_propagation = 3
2019/01/23 11:21:42 [debug] dns_error_ttl = 1
2019/01/23 11:21:42 [debug] dns_hostsfile = "/etc/hosts"
2019/01/23 11:21:42 [debug] dns_no_sync = false
2019/01/23 11:21:42 [debug] dns_not_found_ttl = 30
2019/01/23 11:21:42 [debug] dns_order = {"LAST","SRV","A","CNAME"}
2019/01/23 11:21:42 [debug] dns_resolver = {}
2019/01/23 11:21:42 [debug] dns_stale_ttl = 4
2019/01/23 11:21:42 [debug] error_default_type = "text/plain"
2019/01/23 11:21:42 [debug] headers = {"server_tokens","latency_tokens"}
2019/01/23 11:21:42 [debug] log_level = "notice"
2019/01/23 11:21:42 [debug] lua_package_cpath = ""
2019/01/23 11:21:42 [debug] lua_package_path = "./?.lua;./?/init.lua;"
2019/01/23 11:21:42 [debug] lua_socket_pool_size = 30
2019/01/23 11:21:42 [debug] lua_ssl_verify_depth = 1
2019/01/23 11:21:42 [debug] mem_cache_size = "128m"
2019/01/23 11:21:42 [debug] nginx_admin_directives = {}
2019/01/23 11:21:42 [debug] nginx_daemon = "on"
2019/01/23 11:21:42 [debug] nginx_http_directives = {}
2019/01/23 11:21:42 [debug] nginx_optimizations = true
2019/01/23 11:21:42 [debug] nginx_proxy_directives = {}
2019/01/23 11:21:42 [debug] nginx_user = "nobody nobody"
2019/01/23 11:21:42 [debug] nginx_worker_processes = "auto"
2019/01/23 11:21:42 [debug] origins = {}
2019/01/23 11:21:42 [debug] pg_database = "kong"
2019/01/23 11:21:42 [debug] pg_host = "127.0.0.1"
2019/01/23 11:21:42 [debug] pg_password = "******"
2019/01/23 11:21:42 [debug] pg_port = 5432
2019/01/23 11:21:42 [debug] pg_ssl = false
2019/01/23 11:21:42 [debug] pg_ssl_verify = false
2019/01/23 11:21:42 [debug] pg_timeout = 5000
2019/01/23 11:21:42 [debug] pg_user = "kong"
2019/01/23 11:21:42 [debug] plugins = {"bundled"}
2019/01/23 11:21:42 [debug] prefix = "/usr/local/kong/"
2019/01/23 11:21:42 [debug] proxy_access_log = "logs/access.log"
2019/01/23 11:21:42 [debug] proxy_error_log = "logs/error.log"
2019/01/23 11:21:42 [debug] proxy_listen = {"0.0.0.0:8090","0.0.0.0:8443 ssl"}
2019/01/23 11:21:42 [debug] real_ip_header = "X-Real-IP"
2019/01/23 11:21:42 [debug] real_ip_recursive = "off"
2019/01/23 11:21:42 [debug] ssl_cipher_suite = "modern"
2019/01/23 11:21:42 [debug] ssl_ciphers = "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256"
2019/01/23 11:21:42 [debug] stream_listen = {"off"}
2019/01/23 11:21:42 [debug] trusted_ips = {}
2019/01/23 11:21:42 [debug] upstream_keepalive = 60
2019/01/23 11:21:42 [verbose] prefix in use: /usr/local/kong
2019/01/23 11:21:42 [debug] loading subsystems migrations...
2019/01/23 11:21:42 [verbose] retrieving keyspace schema state...
2019/01/23 11:21:42 [verbose] schema state retrieved
2019/01/23 11:21:42 [info] bootstrapping database...
2019/01/23 11:21:42 [debug] creating 'kong' keyspace if not existing...
2019/01/23 11:21:42 [debug] successfully created 'kong' keyspace
2019/01/23 11:21:42 [debug] creating 'schema_meta' table if not existing...
2019/01/23 11:21:42 [debug] successfully created 'schema_meta' table
2019/01/23 11:21:42 [debug] creating 'locks' table if not existing...
2019/01/23 11:21:42 [debug] successfully created 'locks' table
2019/01/23 11:21:42 [verbose] waiting for Cassandra schema consensus (60000ms timeout)...
2019/01/23 11:21:43 [verbose] Cassandra schema consensus: reached
2019/01/23 11:21:43 [info] migrating core on keyspace 'kong'...
2019/01/23 11:21:43 [debug] running migration: 000_base
2019/01/23 11:21:46 [info] core migrated up to: 000_base (executed)
2019/01/23 11:21:46 [debug] running migration: 001_14_to_15
2019/01/23 11:21:50 [info] core migrated up to: 001_14_to_15 (executed)
2019/01/23 11:21:50 [debug] running migration: 002_15_to_1
2019/01/23 11:21:50 [info] core migrated up to: 002_15_to_1 (executed)
2019/01/23 11:21:50 [info] migrating oauth2 on keyspace 'kong'...
2019/01/23 11:21:50 [debug] running migration: 000_base_oauth2
2019/01/23 11:21:52 [info] oauth2 migrated up to: 000_base_oauth2 (executed)
2019/01/23 11:21:52 [debug] running migration: 001_14_to_15
Error:
/usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:75: [Cassandra error] cluster_mutex callback threw an error: /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:68: [Cassandra error] failed to run migration '001_14_to_15' teardown: [Read timeout] Operation timed out - received only 0 responses.
stack traceback:
        [C]: in function 'assert'
        /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:68: in function </usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:67>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/db/init.lua:390: in function 'cluster_mutex'
        /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:67: in function 'bootstrap'
        /usr/local/share/lua/5.1/kong/cmd/migrations.lua:114: in function 'cmd_exec'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:87>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:44>
        /usr/local/bin/kong:7: in function 'file_gen'
        init_worker_by_lua:50: in function <init_worker_by_lua:48>
        [C]: in function 'xpcall'
        init_worker_by_lua:57: in function <init_worker_by_lua:55>
stack traceback:
        [C]: in function 'error'
        /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:75: in function 'bootstrap'
        /usr/local/share/lua/5.1/kong/cmd/migrations.lua:114: in function 'cmd_exec'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:87>
        [C]: in function 'xpcall'
        /usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:44>
        /usr/local/bin/kong:7: in function 'file_gen'
        init_worker_by_lua:50: in function <init_worker_by_lua:48>
        [C]: in function 'xpcall'
        init_worker_by_lua:57: in function <init_worker_by_lua:55>

**Comment by JustUse** [comment]
> Strange, I am just a community member but I have never been able to reproduce that. You will probably want to run in --vv mode and also give Kong team(I am just a community member) all your C* configurations and connection settings. Something seems off here.

[root@localhost kong]# kong migrations bootstrap --db-timeout 60 --vv
2019/01/23 11:21:42 [verbose] Kong: 1.0.0
2019/01/23 11:21:42 [debug] ngx_lua: 10013
2019/01/23 11:21:42 [debug] nginx: 1013006
2019/01/23 11:21:42 [debug] Lua: LuaJIT 2.1.0-beta3
2019/01/23 11:21:42 [verbose] reading config file at /etc/kong/kong.conf
2019/01/23 11:21:42 [debug] reading environment variables
2019/01/23 11:21:42 [debug] admin_access_log = "logs/admin_access.log"
2019/01/23 11:21:42 [debug] admin_error_log = "logs/error.log"
2019/01/23 11:21:42 [debug] admin_listen = {"0.0.0.0:8001","0.0.0.0:8444 ssl"}
2019/01/23 11:21:42 [debug] anonymous_reports = false
2019/01/23 11:21:42 [debug] cassandra_consistency = "ONE"
2019/01/23 11:21:42 [debug] cassandra_contact_points = {"192.168.0.119","192.168.0.146"}
2019/01/23 11:21:42 [debug] cassandra_data_centers = {"dc1:2","dc2:3"}
2019/01/23 11:21:42 [debug] cassandra_keyspace = "kong"
2019/01/23 11:21:42 [debug] cassandra_lb_policy = "RequestRoundRobin"
2019/01/23 11:21:42 [debug] cassandra_port = 9042
2019/01/23 11:21:42 [debug] cassandra_repl_factor = 1
2019/01/23 11:21:42 [debug] cassandra_repl_strategy = "SimpleStrategy"
2019/01/23 11:21:42 [debug] cassandra_schema_consensus_timeout = 20000
2019/01/23 11:21:42 [debug] cassandra_ssl = false
2019/01/23 11:21:42 [debug] cassandra_ssl_verify = false
2019/01/23 11:21:42 [debug] cassandra_timeout = 10000
2019/01/23 11:21:42 [debug] cassandra_username = "kong"
2019/01/23 11:21:42 [debug] client_body_buffer_size = "8k"
2019/01/23 11:21:42 [debug] client_max_body_size = "0"
2019/01/23 11:21:42 [debug] client_ssl = false
2019/01/23 11:21:42 [debug] database = "cassandra"
2019/01/23 11:21:42 [debug] db_cache_ttl = 0
2019/01/23 11:21:42 [debug] db_resurrect_ttl = 30
2019/01/23 11:21:42 [debug] db_update_frequency = 5
2019/01/23 11:21:42 [debug] db_update_propagation = 3
2019/01/23 11:21:42 [debug] dns_error_ttl = 1
2019/01/23 11:21:42 [debug] dns_hostsfile = "/etc/hosts"
2019/01/23 11:21:42 [debug] dns_no_sync = false
2019/01/23 11:21:42 [debug] dns_not_found_ttl = 30
2019/01/23 11:21:42 [debug] dns_order = {"LAST","SRV","A","CNAME"}
2019/01/23 11:21:42 [debug] dns_resolver = {}
2019/01/23 11:21:42 [debug] dns_stale_ttl = 4
2019/01/23 11:21:42 [debug] error_default_type = "text/plain"
2019/01/23 11:21:42 [debug] headers = {"server_tokens","latency_tokens"}
2019/01/23 11:21:42 [debug] log_level = "notice"
2019/01/23 11:21:42 [debug] lua_package_cpath = ""
2019/01/23 11:21:42 [debug] lua_package_path = "./?.lua;./?/init.lua;"
2019/01/23 11:21:42 [debug] lua_socket_pool_size = 30
2019/01/23 11:21:42 [debug] lua_ssl_verify_depth = 1
2019/01/23 11:21:42 [debug] mem_cache_size = "128m"
2019/01/23 11:21:42 [debug] nginx_admin_directives = {}
2019/01/23 11:21:42 [debug] nginx_daemon = "on"
2019/01/23 11:21:42 [debug] nginx_http_directives = {}
2019/01/23 11:21:42 [debug] nginx_optimizations = true
2019/01/23 11:21:42 [debug] nginx_proxy_directives = {}
2019/01/23 11:21:42 [debug] nginx_user = "nobody nobody"
2019/01/23 11:21:42 [debug] nginx_worker_processes = "auto"
2019/01/23 11:21:42 [debug] origins = {}
2019/01/23 11:21:42 [debug] pg_database = "kong"
2019/01/23 11:21:42 [debug] pg_host = "127.0.0.1"
2019/01/23 11:21:42 [debug] pg_password = "******"
2019/01/23 11:21:42 [debug] pg_port = 5432
2019/01/23 11:21:42 [debug] pg_ssl = false
2019/01/23 11:21:42 [debug] pg_ssl_verify = false
2019/01/23 11:21:42 [debug] pg_timeout = 5000
2019/01/23 11:21:42 [debug] pg_user = "kong"
2019/01/23 11:21:42 [debug] plugins = {"bundled"}
2019/01/23 11:21:42 [debug] prefix = "/usr/local/kong/"
2019/01/23 11:21:42 [debug] proxy_access_log = "logs/access.log"
2019/01/23 11:21:42 [debug] proxy_error_log = "logs/error.log"
2019/01/23 11:21:42 [debug] proxy_listen = {"0.0.0.0:8090","0.0.0.0:8443 ssl"}
2019/01/23 11:21:42 [debug] real_ip_header = "X-Real-IP"
2019/01/23 11:21:42 [debug] real_ip_recursive = "off"
2019/01/23 11:21:42 [debug] ssl_cipher_suite = "modern"
2019/01/23 11:21:42 [debug] ssl_ciphers = "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256"
2019/01/23 11:21:42 [debug] stream_listen = {"off"}
2019/01/23 11:21:42 [debug] trusted_ips = {}
2019/01/23 11:21:42 [debug] upstream_keepalive = 60
2019/01/23 11:21:42 [verbose] prefix in use: /usr/local/kong
2019/01/23 11:21:42 [debug] loading subsystems migrations...
2019/01/23 11:21:42 [verbose] retrieving keyspace schema state...
2019/01/23 11:21:42 [verbose] schema state retrieved
2019/01/23 11:21:42 [info] bootstrapping database...
2019/01/23 11:21:42 [debug] creating 'kong' keyspace if not existing...
2019/01/23 11:21:42 [debug] successfully created 'kong' keyspace
2019/01/23 11:21:42 [debug] creating 'schema_meta' table if not existing...
2019/01/23 11:21:42 [debug] successfully created 'schema_meta' table
2019/01/23 11:21:42 [debug] creating 'locks' table if not existing...
2019/01/23 11:21:42 [debug] successfully created 'locks' table
2019/01/23 11:21:42 [verbose] waiting for Cassandra schema consensus (60000ms timeout)...
2019/01/23 11:21:43 [verbose] Cassandra schema consensus: reached
2019/01/23 11:21:43 [info] migrating core on keyspace 'kong'...
2019/01/23 11:21:43 [debug] running migration: 000_base
2019/01/23 11:21:46 [info] core migrated up to: 000_base (executed)
2019/01/23 11:21:46 [debug] running migration: 001_14_to_15
2019/01/23 11:21:50 [info] core migrated up to: 001_14_to_15 (executed)
2019/01/23 11:21:50 [debug] running migration: 002_15_to_1
2019/01/23 11:21:50 [info] core migrated up to: 002_15_to_1 (executed)
2019/01/23 11:21:50 [info] migrating oauth2 on keyspace 'kong'...
2019/01/23 11:21:50 [debug] running migration: 000_base_oauth2
2019/01/23 11:21:52 [info] oauth2 migrated up to: 000_base_oauth2 (executed)
2019/01/23 11:21:52 [debug] running migration: 001_14_to_15
Error:
/usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:75: [Cassandra error] cluster_mutex callback threw an error: /usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:68: [Cassandra error] failed to run migration '001_14_to_15' teardown: [Read timeout] Operation timed out - received only 0 responses.
stack traceback:
[C]: in function 'assert'
/usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:68: in function </usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:67>
[C]: in function 'xpcall'
/usr/local/share/lua/5.1/kong/db/init.lua:390: in function 'cluster_mutex'
/usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:67: in function 'bootstrap'
/usr/local/share/lua/5.1/kong/cmd/migrations.lua:114: in function 'cmd_exec'
/usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:87>
[C]: in function 'xpcall'
/usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:44>
/usr/local/bin/kong:7: in function 'file_gen'
init_worker_by_lua:50: in function <init_worker_by_lua:48>
[C]: in function 'xpcall'
init_worker_by_lua:57: in function <init_worker_by_lua:55>
stack traceback:
[C]: in function 'error'
/usr/local/share/lua/5.1/kong/cmd/utils/migrations.lua:75: in function 'bootstrap'
/usr/local/share/lua/5.1/kong/cmd/migrations.lua:114: in function 'cmd_exec'
/usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:87>
[C]: in function 'xpcall'
/usr/local/share/lua/5.1/kong/cmd/init.lua:87: in function </usr/local/share/lua/5.1/kong/cmd/init.lua:44>
/usr/local/bin/kong:7: in function 'file_gen'
init_worker_by_lua:50: in function <init_worker_by_lua:48>
[C]: in function 'xpcall'
init_worker_by_lua:57: in function <init_worker_by_lua:55>

**Comment by thibaultcha** [comment]
This feels similar to #4229, which encountered a slightly more descriptive error. I notice that the migration in question in the issue at hand here is also assuming that the schema is up-to-date in between its `up` and `teardown` step.

**Comment by JustUse** [comment]
> This feels similar to #4229, which encountered a slightly more descriptive error. I notice that the migration in question in the issue at hand here is also assuming that the schema is up-to-date in between its `up` and `teardown` step.

ok, but after migration finish with that errors, caused the rate-limiting failed.

**Comment by thibaultcha** [comment]
Considering this resolved thanks to #4233. It was most likely the same root cause as #4229. If you happen to run another `bootstrap` with Kong 1.0.3 that fails, please let us know in another issue, thanks!

**Comment by JustUse** [comment]
> Considering this resolved thanks to #4233. It was most likely the same root cause as #4229. If you happen to run another `bootstrap` with Kong 1.0.3 that fails, please let us know in another issue, thanks!

It was ok when i run 'bootstrap' with Kong 1.0.3, thanks!
